mitodl/micromasters
# Implemented sidenav

#### What are the relevant tickets?

this closes #1414 

#### What's this PR do?

This implements a side drawer navigation for mobile devices, following the mockups.

#### How should this be manually tested?

You can either use the pr review app (https://micromasters-ci-pr-1460.herokuapp.com/) on a mobile device or use the device simulator in Chrome or similar to check out the mobile view. 

Basically, you should be able to navigate around the site using the sidebar.

you should be able to:

- click the hamburger to open the sidebar
- visit different links
- navigate as usual on the desktop view of the app

the sidebar and mobile nav stuff should:

- not be visible when using a desktop-size (> 900px wide) view
- be dismissable by either clicking outside of the navbar or by swiping to the left on it
- hide itself after you click a link
- hide the program switch on the `/learner`, `/profile`, and `/settings` pages

the desktop nav stuff:

- should be hidden on a mobile view
- should have no regressions or strangeness


#### Any background context you want to provide?

#### Screenshots (if appropriate)

![closed](https://cloud.githubusercontent.com/assets/6207644/19602192/47019bea-977a-11e6-9134-4570e7e1444e.png)

![open](https://cloud.githubusercontent.com/assets/6207644/19602193/4a25ca30-977a-11e6-92a5-35b4af0c29e5.png)


#### What GIF best describes this PR or how it makes you feel?
